File: 0338-counting-bits/0338-counting-bits.py
class Solution:   
    def countBits(self, n: int) -> List[int]:
        def ctob(n:int)->int:
            rem,re=0,0
            while(n!=0):
                rem=n%2
                n=n//2
                re=re+rem
            return re
                
            
        res=[]
        for i in range(n+1):
            t=int(ctob(i))
            res.append(t)
        return res
            
        
        

File: 0338-counting-bits/0338-counting-bits.cpp
class Solution {
public:
    int cdtb(int n){
        int rem=0,res=0;
        
        while(n!=0){
            rem=n%2;
            n=n/2;
            res=res+rem;
        }
       return res; 
    }
    vector<int> countBits(int n) {
        vector<int> ret;
        for(int i=0;i<=n;i++){
            int no1=cdtb(i);
            ret.push_back(no1);
        }
        return ret;
    }
};